salt/client/ssh/__init__.py

```diff
@@ -1106,7 +1106,10 @@ def shim_cmd(self, cmd_str, extension='py'):
             shim_tmp_file.write(salt.utils.to_bytes(cmd_str))
 
         # Copy shim to target system, under $HOME/.<randomized name>
-        target_shim_file = '.{0}.{1}'.format(binascii.hexlify(os.urandom(6)), extension)
+        target_shim_file = '.{0}.{1}'.format(
+            binascii.hexlify(os.urandom(6)).decode('ascii'),
```

Comment:
You likely only want to decode under py3 right?

Reply:
Good catch. Decoding on python 2 doesn't cause an issue because the unicode string is decoded by format.

```
>>> '{}'.format(binascii.hexlify(os.urandom(6)).decode('ascii'))
'1f84d3dc6284'
```